CACHING.md

```diff
@@ -0,0 +1,143 @@
+# Cache Recipes
+
+This document focuses on gathering all the common use cases that will help the users of `actions/cache`
+- Know the use cases that can be tackled and sample workflows on how to tackle them
+- Optimise their workflows to use the cache inputs and outputs better
+
+## Actions Cache Basics
+
+The cache action works using the following set of inputs and outputs as mentioned [here](https://github.com/actions/cache#inputs). However these inputs are self explaining below are some ways in which the inputs can be used in a better way.
+
+### Keys
+
+#### Static keys
+
+```yaml
+    - uses: actions/cache@v3
+      with:
+        key: static-key
+```
+
+In your workflows, you can use keys in a hardcoded manner. This way the same `key` will be saved once and restored till its evicted. In case the `key` gets evicted, cache with same `key` will be created again and saved.
+
+#### Dynamic keys
+
+```yaml
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    - uses: actions/cache@v3
+      with:
+        key: ${{ runner.os }}-cache
+```
+
+In your workflows, if you wish to create OS specific caches, or caches based on the lockfiles, commit SHA, workflow run id, etc. then you can generate the keys dynamically at run-time. Below are some of the ways to use dynamically generated keys
+
+Cache key by Operating system: 
+```yaml
+    key: ${{ runner.os }}-cache
+```
+Cache key by Workflow run id/attempt: 
+```yaml
+    key: cache-${{ github.run_id }}-${{ github.run_attempt }}
+```
+Cache key by commit id: 
+```yaml
+    key: cache-${{ github.sha }}
+```
+Cache key by lockfile: 
+```yaml
+    key: cache-${{ hashFiles('**/lockfiles') }}
+```
+Cache key by combination of multiple options: 
+```yaml
+    key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
+```
+
+The [GitHub Context](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) can be used to create keys using the workflows metadata.
+
+### Paths
```

Comment:
I don't think we should explain what is Path. We should tell that a selected list of paths can be used. For example for node, we can cache npm-cache directory. Similarly few more examples.